### ISSUE TITLE
New bug found - Mouse pointer not lining up in touch/absolute mode

### ISSUE THREAD

**Issue by jevansoh** [libretro/desmume#31]
Hello,

I have been working with @wertz over on the libretro forum and you can find that thread at https://forums.libretro.com/t/desmume-core-mouse-very-erratic/15318/82 

I'm offering a $50 bounty/payment via PayPal to anyone who can help me resolve a new bug we found, at least on my system.

First, when the option (quick menu, options, in RetroArch) is set to mouse/relative, the mouse movement is very fast (even with mouse speed on the lowest setting) and erratic.

I would rather use touch/absolute mode anyway, though, but the problem is, on the newest version of the core (and older versions of the same core) I have to click about 2 to 3 inches above where I actually want to click in order for it to work/register my click.  This is on a 27" monitor at 1920x1080 resolution.  Obviously, this is guesswork and you can't play a game like this.

I found an older version of a core called Desmume_Future and when setting it to absolute mode, it works flawlessly.  The problem is, it doesn't have a screen gap option, so I can't use that core.

For many more details and all the troubleshooting we did plus more info on the problem itself, please visit my thread listed above.

We've worked on this for hours and hours and there are 83 posts, but it should convey the problem more clearly and hopefully a solution can be found.

I know nothing about programming, but it seems to me it might be easy to fix because if you could just copy the portion of code out of the Desmume_Future core (having to do with using the mouse in absolute mode) to the new core, the problem would be solved since it works great in the "Future" core.

I hope you understand the problem, as it is kind of difficult to explain and I would greatly appreciate some help and a resolution to this problem.

Thanks so much,

--Jason

**Comment by bearoso** [comment]
Should be fixed in ab6a763. Also added some slower mouse speed options.

**Comment by jevansoh** [comment]
Thanks again for doing this for me.  I really appreciate it.  How do I go about turning this "cpp" file into a new dll core I can use?

I'm sorry, but I'm totally new to all of this and have no clue what I'm doing or where to begin. :)

Be sure to send me your address for payment.

Thanks again,

--J

**Comment by bearoso** [comment]
Just wait until the buildbot gets around to compiling the source. It hasn’t yet, so it could be a day or so. Then use the Online updater->Core updater in the menu to redownload the core and get the new version. No payment necessary, just let us know if it’s not fixed.

**Comment by jevansoh** [comment]
Oh, I see.  At least I'm not missing something. :)  You're kidding?  You did all that work just for me and you don't want payment?  Thank you SO MUCH!  That is so very kind of you.  You've seriously made my day.

I'll check tonight and see if there is a new version from the online updater and will report back once there is and let you know if it's fixed.

Thanks again!!!

--J

**Comment by jevansoh** [comment]
All fixed!!! WORKS GREAT!

Thank you all who replied and helped and thanks to the devs and folks that work so hard on this!

I'm off to play NDS now!

--J

**Comment by jevansoh** [comment]
Unfortunately, I'm having one last remaining problem...

Audio Mute is on.  I turn it off, and then I save it in settings, save current config, and I've tried save new config, and even save core overrides, but as soon as I exit a game and go back in, whether the same game or a new game, audio mute is back on again.

I can't get it to stay off.

I can turn it off then play the game with the audio on, but no matter how I save it, as soon as I exit and go back in to play again, audio mute is on again.

What can I do about this?

Thanks,

--J

**Comment by bearoso** [comment]
You’ve probably got an override somewhere that’s overwriting the settings. Go into the Appdata\Roaming\RetroArch folder in your user directory and erase any files like desmume.cfg or desmume-libretro.cfg or any .cfg named after your games in that folder or the config subfolder.

When you start up again the audio may be muted, but unmute it and avoid saving core override or game override again.

**Comment by jevansoh** [comment]
I figured it out, but am still curious... I noticed it will save a new config but when I try to save current config, it says "failed saving current config" and I don't know why.  I deleted the desmume.cfg and replaced it with desmume-7.cfg (I had tried 7 times to save it, I guess) and that worked, but I wonder why it won't save the current config?

Anyway...Everything seems to be working now.

Thanks again,

--J
